pkg/crc/machine/machine.go

```diff
@@ -253,6 +253,14 @@ func Start(startConfig StartConfig) (StartResult, error) {
 		}
 	}
 
+	// Generate and copy new ssh key to the newly created vm
+	if !exists {
+		logging.Info("Generating new SSH key")
+		if err := updateSSHKeyPair(sshRunner); err != nil {
+			return startError(startConfig.Name, "Error updating public key", err)
+		}
+	}
+
```

Comment:
Then it should be after L#255, as soon as ssh is available.

Reply:
Aha (its as soon as possible :-), we are only doing these two things after the start. i think it makes sense to change the ssh key first and then send the kubeconfig.

This would obviously not prevent the start failures, but at least we'll know why a second start fails when the initial start failed, now that the potential missing `id_rsa` and `kubeconfig` errors should be avoided, we might see the real error why it fails.